lib/std/array_list.zig

```diff
@@ -663,16 +750,36 @@ pub fn ArrayListAlignedUnmanaged(comptime T: type, comptime alignment: ?u29) typ
             self.items[n] = item;
         }
 
-        /// Insert slice `items` at index `i`. Moves `list[i .. list.len]` to
-        /// higher indicices make room.
+        /// Resize the array, adding `count` new elements at position `index`, which have `undefined` values.
+        /// The return value is a slice pointing to the newly allocated elements. The returned pointer
+        /// becomes invalid when the list is resized. Resizes list if self.capacity is not large enough.
+        pub fn addManyAtIndex(
+            self: *Self,
+            allocator: Allocator,
+            index: usize,
+            count: usize,
+        ) Allocator.Error![]T {
+            var managed = self.toManaged(allocator);
+            const result = try managed.addManyAtIndex(index, count);
+            self.* = managed.moveToUnmanaged();
+            return result;
```

Comment:
This code is incorrect, as is `replaceRange` in master branch. It should be:

```suggestion
            var managed = self.toManaged(allocator);
            defer self.* = managed.moveToUnmanaged();
            return managed.addManyAtIndex(index, count);
```

Otherwise, any state mutations that occur before returning an error code are lost (such as reallocating the buffer).

It looks like since there's only 1 `try` in each of these implementations, we can get away with it, but really it should be written more carefully as above.